src/settings.ts

```diff
@@ -0,0 +1,98 @@
+                }
+                if (debugSettingsRoot.stepping.skipSynthetics) {
+                    stepFilters["skipSynthetics"] = debugSettingsRoot.stepping.skipSynthetics;
+                }
+                if (debugSettingsRoot.stepping.skipStaticInitializers) {
+                    stepFilters["skipStaticInitializers"] = debugSettingsRoot.stepping.skipStaticInitializers;
+                }
+                if (debugSettingsRoot.stepping.skipConstructors) {
+                    stepFilters["skipConstructors"] = debugSettingsRoot.stepping.skipConstructors;
+                }
+                extraSettings["stepFilters"] = stepFilters;
+            }
+            if (debugSettingsRoot.exceptionBreakpoint && Object.keys(debugSettingsRoot.exceptionBreakpoint).length) {
+                let exceptionFilters = {};
+                if (debugSettingsRoot.exceptionBreakpoint.exceptionTypes) {
+                    exceptionFilters["exceptionTypes"] = debugSettingsRoot.exceptionBreakpoint.exceptionTypes;
+                }
+                if (debugSettingsRoot.exceptionBreakpoint.allowClasses) {
+                    exceptionFilters["allowClasses"] = debugSettingsRoot.exceptionBreakpoint.allowClasses;
+                }
+                if (debugSettingsRoot.exceptionBreakpoint.skipClasses) {
+                    exceptionFilters["skipClasses"] = await substituteFilterVariables(debugSettingsRoot.exceptionBreakpoint.skipClasses);
+                }
+                extraSettings["exceptionFilters"] = exceptionFilters;
+                extraSettings["exceptionFiltersUpdated"] = true;
+            }
+
+            if (debugSettingsRoot.jdwp) {
+                if (debugSettingsRoot.jdwp.async) {
+                    extraSettings["asyncJDWP"] = debugSettingsRoot.jdwp.async;
+                }
+                if (debugSettingsRoot.jdwp.limitOfVariablesPerJdwpRequest) {
+                    extraSettings["limitOfVariablesPerJdwpRequest"] = Math.max(debugSettingsRoot.jdwp.limitOfVariablesPerJdwpRequest, 1);
+                }
+                if (debugSettingsRoot.jdwp.requestTimeout) {
+                    extraSettings["jdwpRequestTimeout"] = Math.max(debugSettingsRoot.jdwp.requestTimeout, 100);
+                }
+            }
+            const settings = await commands.executeCommand(Commands.JAVA_UPDATE_DEBUG_SETTINGS, JSON.stringify(
+                {
+                    ...debugSettingsRoot,
+                    ...extraSettings,
+                    logLevel,
+                }));
+            if (logLevel === "FINE") {
+                // tslint:disable-next-line:no-console
+                console.log("settings:", settings);
+            }
+        } catch (err) {
+            // log a warning message and continue, since update settings failure should not block debug session
+            // tslint:disable-next-line:no-console
+            console.log("Cannot update debug settings.", err);
```

Comment:
I know this was copied from vscode-java-debug, buy let's use console.error here instead.
```suggestion
            console.error("Cannot update debug settings.", err);
```